### PR TITLE
backend: (riscv) print assembly from riscv_func, remove region from label

### PR DESCRIPTION
**Pull request by superlopuh**
I don't think we need the extra step of rewriting to labels, and we benefit from skipping it. For one, fewer rewrites means fewer opportunities for bugs. Also, we get fewer `GetRegisterOp`s, and having 0 of them lets us interpret the code directly. As discussed before, dropping the region from labels also makes the code a bit less weird/confusing.

### CODE REVIEW

File: docs/Toy/toy/rewrites/setup_riscv_pass.py [hunk 1]
```diff
@@ -26,11 +26,8 @@ def match_and_rewrite(self, op: ModuleOp, rewriter: PatternRewriter):
             ),
         )
         data_section = riscv.AssemblySectionOp(".data", Region(Block()))
-        text_section = riscv.AssemblySectionOp(
-            ".text", rewriter.move_region_contents_to_new_regions(op.regions[0])
```
Comment (superlopuh) [inline docs/Toy/toy/rewrites/setup_riscv_pass.py:30]: The ".text" assembler directive is now inserted before every function in xDSL so this one is not needed

File: tests/dialects/test_riscv.py [hunk 2]
```diff
@@ -109,22 +108,6 @@ def test_label_op_with_comment():
     assert code == f"{label_str}:                                         # my label\n"
 
 
-def test_label_op_with_region():
```
Comment (superlopuh) [inline tests/dialects/test_riscv.py:112]: labels no longer have regions, so this test is not needed

File: tests/filecheck/dialects/riscv/riscv_ops.mlir [hunk 3]
```diff
@@ -397,6 +397,6 @@
 // CHECK-GENERIC-NEXT:     %fmv_w_x = "riscv.fmv.w.x"(%0) : (!riscv.reg<>) -> !riscv.freg<>
 // CHECK-GENERIC-NEXT:     %flw = "riscv.flw"(%0) {"immediate" = 1 : si12} : (!riscv.reg<>) -> !riscv.freg<>
 // CHECK-GENERIC-NEXT:     "riscv.fsw"(%0, %f0) {"immediate" = 1 : si12} : (!riscv.reg<>, !riscv.freg<>) -> ()
-// CHECK-GENERIC-NEXT:     "riscv.ret"() : () -> ()
-// CHECK-GENERIC-NEXT:   }) {"label" = #riscv.label<"main">} : () -> ()
+// CHECK-GENERIC-NEXT:     "riscv_func.return"() : () -> ()
+// CHECK-GENERIC-NEXT:   }) {"sym_name" = "main", "function_type" = () -> ()} : () -> ()
```
Comment (kingiler) [inline tests/filecheck/dialects/riscv/riscv_ops.mlir:401]: Just curious why we need `"function_type" ` attributes in `FuncOp`.

**Review by kingiler — APPROVED** [review]
Thank you! It makes code much easier to read.

File: tests/filecheck/dialects/riscv/riscv_ops.mlir (hunk 3 above)
Comment (superlopuh) [reply tests/filecheck/dialects/riscv/riscv_ops.mlir:401]: We copied this over from `func.func`, my understanding is that it's useful for external function declarations, among other things. At one point we might want to adopt the FunctionOpInterface from MLIR and it's a requirement there AFAIK. https://mlir.llvm.org/docs/Dialects/Func/#funcfunc-funcfuncop

**Review by nazavode — APPROVED** [review]
> As discussed before, dropping the region from labels also makes the code a bit less weird/confusing.

Nice, thanks.

**Review by adutilleul — APPROVED** [review]

**Review by compor — APPROVED** [review]

**Pull request merged**